### PR TITLE
Fixed problem with DST

### PR DESCRIPTION
**Pull request by tompazourek**
tl;dr **We should be looking at the time zone offset at the target moment, not the time zone offset at current moment. Even though we use the same time zone, the offsets changes depending on the date (depends on DST).**

The offset computation of `moment().tz(options.timeZone)` code used in the `getMoment` function can refer to wrong time zone offset if the current time zone and is different from the timezone in the target moment.

This can happen for example if the current time (aka `moment()`) is in the DST and the target time (aka `moment(d, parseFormats, options.useStrict)`) is not in the DST.

### CODE REVIEW

**Comment by kyranb** [comment]
@Eonasdan Any chance of getting this merged? I'm unable to use the 'stable' version as this breaks its functionality completely.

**Comment by kyranb** [comment]
Any update on this?
